### PR TITLE
Judges | Assign cases | Select cases

### PR DESCRIPTION
**Pull request by evankroske**
Connects https://github.com/department-of-veterans-affairs/caseflow/issues/4716

### Description
Allows judges to select cases and fixes some regressions.

### Testing Plan
- [x] Log in as BVAAABSHIRE.
- [x] On /queue, switch to assign cases.
- [x] Select Evangeline.
- [x] Click on Ezra.
- [x] On Unassigned Cases, check that Evangeline is still be selected.
- [x] Log in as BVARERDMAN.
- [x] On /queue, switch to assign cases.
- [x] On Sunny Ritchie, select Tatum.
- [x] Click on Jalyn.
- [x] On Sunny Ritchie, check that Tatum is still selected.

### CODE REVIEW

File: client/app/queue/reducers.js [hunk 1]
```diff
@@ -261,6 +262,18 @@ const workQueueReducer = (state = initialState, action = {}) => {
         }
       }
     });
+  case ACTIONS.SET_SELECTION_OF_TASK_OF_USER:
+    return update(state, {
+      isVacolsIdAssignedToUserSelected: {
+        [action.payload.userId]: {
+          $set: update(state.isVacolsIdAssignedToUserSelected[action.payload.userId] || {}, {
```
Comment (dannysellers) [inline client/app/queue/reducers.js]: I feel like the complexity of this operation outweighs the simplicity of getting cases by id later on. Could we not just store an array of selected case IDs? We only really need to check whether an ID is present in the set.

This is untested, but something like

```javascript
case ACTIONS.SET_SELECTION_OF_TASK_OF_USER: {
  const { userId, vacolsId } = action.payload;
  // potentially: const selectedAppeals = state.appealsByUser[userId] || [];
  const selectedAppeals = state.isVacolsIdAssignedToUserSelected[userId] || [];

  if (selectedAppeals.includes(vacolsId)) {
    const appealIdx = selectedAppeals.indexOf(vacolsId);

    selectedAppeals.splice(appealIdx, 1);
  } else {
    selectedAppeals.push(vacolsId);
  }

  return update(state, {
    isVacolsIdAssignedToUserSelected: {
      [userId]: {
        $set: selectedAppeals
      }
    }
  });
}
```

File: client/app/queue/JudgeAssignTaskListView.jsx [hunk 2]
```diff
@@ -35,17 +35,20 @@ const UnassignedCasesPage = ({ tasksWithAppeals }) => {
...
       <h2>Unassigned Cases</h2>
-      <JudgeAssignTaskTable tasksAndAppeals={tasksWithAppeals} />
+      <JudgeAssignTaskTable {...props} />
     </React.Fragment>;
   }
 
   return tableContent;
 };
 
 const AssignedCasesPage = connect(
```
Comment (dannysellers) [inline client/app/queue/JudgeAssignTaskListView.jsx]: Nit: This functional component is starting to get a bit unruly. I wonder if we should make a folder inside `queue/components` for these two JudgeAssign views (unrelated: I feel like `CaseDetailsLink` and `AppealDocumentCount` should also probably be moved into `queue/components`)

File: client/app/queue/JudgeAssignTaskTable.jsx [hunk 3]
```diff
@@ -7,13 +7,24 @@ import moment from 'moment';
...
   getCaseDetailsLink = ({ task, appeal }) => <CaseDetailsLink task={task} appeal={appeal} />;
 
   getQueueColumns = () => [
+    {
+      header: COPY.JUDGE_QUEUE_TABLE_SELECT_COLUMN_TITLE,
+      valueFunction:
+        ({task}) => <Checkbox
+          name={task.vacolsId}
+          label="&nbsp;"
+          value={this.props.isVacolsIdSelected[task.vacolsId]}
```
Comment (dannysellers) [inline client/app/queue/JudgeAssignTaskTable.jsx]: `this.props.isVacolsIdSelected[task.vacolsId]` starts out as `undefined`, such that when you check the box for the first time, React throws a warning about an uncontrolled input being changing to be controlled. Setting a default value solves this (i.e. `this.props.isVacolsIdSelected[task.vacolsId] || false`)

File: client/app/queue/JudgeAssignTaskTable.jsx [hunk 4]
```diff
@@ -7,13 +7,24 @@ import moment from 'moment';
...
 
   getCaseDetailsLink = ({ task, appeal }) => <CaseDetailsLink task={task} appeal={appeal} />;
 
   getQueueColumns = () => [
+    {
+      header: COPY.JUDGE_QUEUE_TABLE_SELECT_COLUMN_TITLE,
+      valueFunction:
+        ({task}) => <Checkbox
+          name={task.vacolsId}
+          label="&nbsp;"
```
[the remark this answers is not in the document]
Comment (evankroske) [reply client/app/queue/JudgeAssignTaskTable.jsx]: ✅

File: client/app/queue/reducers.js (hunk 1 above)
Comment (evankroske) [reply client/app/queue/reducers.js]: Discussed offline

File: client/app/queue/QueueActions.js [hunk 5]
```diff
@@ -182,3 +182,8 @@ export const fetchTasksAndAppealsOfAttorney = (attorneyId) => (dispatch) => {
       error }))
   );
 };
+
+export const setSelectionOfTaskOfUser = (payload) => ({
```
[the remark this answers is not in the document]
Comment (evankroske) [reply client/app/queue/QueueActions.js]: Stopped passing payload as-is

File: client/app/queue/JudgeAssignTaskListView.jsx (hunk 2 above)
Comment (evankroske) [reply client/app/queue/JudgeAssignTaskListView.jsx]: Moved AssignedCasesPage to its own file

**Pull request merged**